Clock.py

```diff
@@ -127,21 +126,29 @@ def announce(s, kob, sender):
     from pykob import config, kob, morse, log
     from distutils.util import strtobool
 
-    log.log("Starting Clock")
-
-    clock_parser = argparse.ArgumentParser(parents=[config.PortOverride, config.SoundtOverride, config.WPMOverride])
-    clock_parser.add_argument("-b", "--begin", default=900, type=int, help="Beginning of time announcements ", metavar="time", dest="Begin")
-    clock_parser.add_argument("-e", "--end", default=2200, type=int, help="End of time announcements ", metavar="time", dest="End")
-    clock_parser.add_argument("-i", "--interval", default=60, type=int, help="The time announcement interval in minutes", metavar="minutes", dest="Interval")
-    clock_parser.add_argument("-t", "--text", action='store_true', default=False, help="Whether to print text locally as it's sent to the sounder", dest="Text")
-    args = clock_parser.parse_args()
+    #log.log("Starting Clock")
+
+    arg_parser = argparse.ArgumentParser(description="Morse Cuckoo Clock", parents=\
+     [\
+      config.serial_port_override, \
+      config.code_type_override, \
+      config.sound_override, \
+      config.sounder_override, \
+      config.spacing_override, \
+      config.min_char_speed_override, \
+      config.text_speed_override])
+    arg_parser.add_argument("-B", "--begin", default=900, type=int, help="Beginning of time announcements ", metavar="time", dest="Begin")
+    arg_parser.add_argument("-E", "--end", default=2200, type=int, help="End of time announcements ", metavar="time", dest="End")
+    arg_parser.add_argument("-I", "--interval", default=60, type=int, help="The time announcement interval in minutes", metavar="minutes", dest="Interval")
+    arg_parser.add_argument("-P", "--print", action='store_true', default=False, help="Whether to print text locally as it's sent to the sounder", dest="Text")
+    args = arg_parser.parse_args()
     
-    port = args.Port # serial port for KOB interface
-    speed = args.Speed  # code speed (words per minute)
-    if (speed < 1) or(speed > 50):
```

Comment:
Why not check this (and bail out) before leaving 'speed' set to a bad value from args.Speed (although I do agree this is the right time to check, vs. Configure.py)

Reply:
I didn't change how this was handled, I just changed the parameter names to match the new names. It does a `sys.exit(1)` if the value is out of range.

Maybe I'm misunderstanding your comment?

Reply:
@pwdirks - Is this good as-is, or are you asking me to change something? Sorry, that I don't completely understand what you are asking/suggesting.

Reply:
[ looks like my original response by e-mail didn't properly get attached here.  Repeating: ]

Hi Ed - I think it’s more a stylistic thing and some of the code didn’t show the pattern well.  Rather than, say

	a = args.a
	if (a < 1) or (a > 50)
	    <print an error and exit>

I’d prefer

	if (args.a < 1) or (args.a > 50)
	    <print an error message and exit>
	a = args.a
	...

and of course the check on args.a could be done immediately following argument parsing, not right where it’s used.  But nothing functionally wrong with the code as-is and no need to change the code or defer the merge.